spatialos-sdk/src/worker/component.rs

```diff
@@ -237,14 +239,15 @@ pub struct ComponentDatabase {
 
 impl ComponentDatabase {
     pub fn new() -> Self {
+        let mut vtables = Vec::new();
+        for table in inventory::iter::<WrappedVTable> {
```

Comment:
How does this work? Is it reflection done at runtime, or is the inventory something generated at compile time? This is really really neat.

Reply:
From https://github.com/dtolnay/inventory: 

> Inventory is built on the ctor crate which provides module initialization functions for Rust similar to __attribute__((constructor)) in C. Each call to inventory::submit! produces a shim that evaluates the given expression and registers it into a registry of its corresponding type. This registration happens dynamically as part of life-before-main for statically linked elements. Elements brought in by a dynamically loaded library are registered at the time that dlopen occurs.